### PR TITLE
Feat auth

### PR DESCRIPTION
**Pull request by abhij1607**
This PR consist of

- router setup
- firebase authentication setup
- firestore setup
- login & signup pages
- added firestore storage for todoList

### CODE REVIEW

File: src/firebase/firestore-requests.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  }
+};
+
+export const updateUserDetails = async (userId, todoList, dispatch) => {
+  try {
+    const usersRef = doc(db, "users", userId);
+    await updateDoc(usersRef, todoList);
+    dispatch({ type: "UPDATE_TODO_LIST", payload: todoList.todoList });
+  } catch (error) {
+    console.log(error);
```
Comment (stutisk) [inline src/firebase/firestore-requests.js:36]: Console can be removed

**Review by stutisk — APPROVED** [review]
Great work
Can be merged

**Review by MaruthiPothuganti — COMMENTED** [review]
Code looks clean and well written. Looks good to merge

File: src/firebase/firestore-requests.js (hunk 1 above)
Comment (abhij1607) [reply src/firebase/firestore-requests.js:36]: thanks for suggestion, I'll add toast later

**Pull request merged**